docs/rule/kotlin_library.soy

```diff
@@ -34,7 +34,12 @@ of the compiled class files and resources.
   {param name: 'srcs' /}
   {param default: '[]' /}
   {param desc}
-  The set of <code>.kt</code> or <code>.kts</code> files to compile for this rule.
+  The set of <code>.kt</code>, <code>.java</code> or <code>.kts</code> files to compile for this rule.
+  If any of the files in this list end in <code>.src.zip</code>,
+  then the entries in the ZIP file that end in <code>.java</code> or <code>.kt</code> will be
+  included as ordinary inputs to compilation. This is common when using
+  a {call buck.java_annotation_processor /} to auto-generate some Java/Kotlin source code that
+  needs to be compiled with some hand-written Java/Kotlin code.
```

Comment:
`This is common when...` is all an implementation detail; someone writing `BUCK` files would never need to use this capability for that purpose.